solutions/quickstart-vpc/variables.tf

```diff
@@ -0,0 +1,94 @@
+# Input variables
+########################################################################################################################
+
+variable "ibmcloud_api_key" {
+  type        = string
+  description = "The IBM Cloud api token"
+  sensitive   = true
+}
+
+variable "prefix" {
+  type        = string
+  description = "Prefix for name of all resource created by this example"
+  validation {
+    error_message = "Prefix must begin and end with a letter and contain only letters, numbers, and - characters."
+    condition     = can(regex("^([A-z]|[a-z][-a-z0-9]*[a-z0-9])$", var.prefix))
+  }
+  default = "dev"
+}
+
+variable "region" {
+  type        = string
+  description = "Region where resources are created"
+}
+
+variable "use_existing_resource_group" {
+  type        = bool
+  description = "Whether to use an existing resource group."
+  default     = false
+}
+
+variable "resource_group_name" {
+  type        = string
+  description = "The name of a new or the existing resource group to provision the cluster. If a prefix input variable is passed, it is prefixed to the value in the `<prefix>-value` format."
+}
+
+variable "resource_tags" {
+  type        = list(string)
+  description = "Optional list of tags to be added to created resources"
+  default     = []
+}
+
+variable "ocp_version" {
+  type        = string
+  description = "Version of the OCP cluster to provision"
+  default     = null
+}
+
+variable "access_tags" {
+  type        = list(string)
+  description = "A list of access tags to apply to the resources created by the module."
+  default     = []
+}
+
+variable "ocp_entitlement" {
+  type        = string
+  description = "Value that is applied to the entitlements for OCP cluster provisioning"
+  default     = null
+}
+
+variable "number_worker_nodes" {
+  type        = number
+  description = "The number of workers to create in the cluster"
+  default     = 2
+}
+
+variable "machine_type" {
+  type        = string
+  description = "Worker node machine type. Use 'ibmcloud ks flavors --zone <zone>' to retrieve the list."
+  default     = "bx2.4x16"
+}
+
+variable "operating_system" {
+  type        = string
+  description = "Allowed OS values are RHEL 8 (REDHAT_8_64) or Red Hat Enterprise Linux CoreOS (RHCOS). RHCOS requires VPC clusters created from 4.15 onwards. Upgraded clusters from 4.14 cannot use RHCOS."
+  default     = "REDHAT_8_64"
+}
+
+variable "vpc_id" {
+  type        = string
+  description = "Id of the VPC instance where this cluster will be provisioned"
+}
+
+variable "existing_cos_id" {
+  type        = string
+  description = "The COS id of an already existing COS instance to use for OpenShift internal registry storage."
+  default     = null
+}
+
+variable "cluster_name" {
+  type        = string
+  description = "The name of the new IBM Cloud OpenShift Cluster. If a `prefix` input variable is specified, it is added to this name in the `<prefix>-value` format."
+  default     = "base-ocp"
```

Comment:
hmm, not sure about adding a default value here - every cluster will end up getting called `dev-base-ocp` and you will get conflicts. Perhaps we make this required?